### PR TITLE
added linux/windows build

### PR DESCRIPTION
**Pull request by linuxseller**
Windows is compiled by default, Linux can be chosen.
Also added `-lm` to link with math library

### CODE REVIEW

File: nob.c [hunk 1]
```diff
@@ -98,17 +102,40 @@ bool build_pov_brain_is_weird()
     return result;
 }
 
+void print_usage(void)
+{
+    fprintf(stderr, "nob [-platform] [platform]\n");
+    fprintf(stderr, "\tplatforms supported: windows, linux\n");
```
Comment (pralkarz) [inline nob.c]:
```suggestion
    nob_log(NOB_INFO, "usage: [./]nob [-platform] [platform]");
    nob_log(NOB_INFO, "platforms supported: Windows, Linux");
```

File: nob.c [hunk 2]
```diff
@@ -98,17 +102,40 @@ bool build_pov_brain_is_weird()
...
+
 int main(int argc, char **argv)
 {
+    bool plaform_windows = true;
+    if(argc!=1){
+        if(strcmp(argv[1], "-platform")==0){
+            if(strcmp(argv[2], "linux")==0){
+                plaform_windows = false;
+            } else if (strcmp(argv[2], "windows")==0){
+                plaform_windows = false;
```
Comment (pralkarz) [inline nob.c]:
```suggestion
                plaform_windows = true;
```

File: nob.c [hunk 3]
```diff
@@ -98,17 +102,40 @@ bool build_pov_brain_is_weird()
...
+            } else {
+                fprintf(stderr, "Error: unsupported OS\n");
```
Comment (pralkarz) [inline nob.c]:
```suggestion
                nob_log(NOB_ERROR, "unsupported platform");
```

File: nob.c [hunk 4]
```diff
@@ -98,17 +102,40 @@ bool build_pov_brain_is_weird()
...
+                print_usage();
+                exit(69);
```
Comment (pralkarz) [inline nob.c]:
```suggestion
                return 1;
```

File: nob.c [hunk 5]
```diff
@@ -98,17 +102,40 @@ bool build_pov_brain_is_weird()
...
+            }
+        } else {
+            print_usage();
+            exit(69);
```
Comment (pralkarz) [inline nob.c]:
```suggestion
            return 1;
```

File: nob.c [hunk 6]
```diff
@@ -98,17 +102,40 @@ bool build_pov_brain_is_weird()
...
+        }
+    }
     NOB_GO_REBUILD_URSELF(argc, argv);
```
Comment (pralkarz) [inline nob.c]: Please move it to the top of the function.

**Review by pralkarz — CHANGES_REQUESTED** [review]
Is `-lm` needed by Linux?

**Comment by linuxseller** [comment]
1. Yes, `-lm` is needed by Linux.
2. Having  inconsistent  `Windows, Linux` and `windows, linux` in program arguments might be confusing for the end user. Which letter case should be chosen?

**Comment by pralkarz** [comment]
2. I agree, let's go with lowercase letters then.

**Comment by linuxseller** [comment]
Done (I guess).

**Comment by pralkarz** [comment]
Thanks for the contribution!

**Pull request merged**